src/Build/Evaluation/Expander.cs

```diff
@@ -3156,57 +3156,64 @@ internal object Execute(object objectInstance, IPropertyProvider<T> properties,
                     }
                     else
                     {
-                        // Execute the function given converted arguments
-                        // The only exception that we should catch to try a late bind here is missing method
-                        // otherwise there is the potential of running a function twice!
-                        try
+                        // First attempt to recognize some well-known functions to avoid binding
+                        // and potential first-chance MissingMethodExceptions
+                        functionResult = ExecuteWellKnownFunction(objectInstance, args);
+
+                        if (functionResult == null)
```

Comment:
What if one of the well-known functions returns null?  Cross that bridge if we come to it?

Reply:
yeah, just fall back to the old one :)

Reply:
I think (maybe?) what Daniel is saying which I caught too is if one of those were to return null as the correct value it would fallback and evaluate again. Could change to `TryExecuteWellKnownFunction` and return a `bool` if the value was found?